### PR TITLE
feature(table): add type generic for filter parameters

### PR DESCRIPTION
**Pull request by TimWieder**
@plentymarkets/team-terra

#### Must be done by Terra

Testing

-   [ ] Person 1 (mandatory)

    Browser-Support (relevant for changes in scss / appearance of components)

    -   [ ] Chrome
    -   [ ] Firefox
    -   [ ] Safari

    Responsiveness

    -   [ ] Desktop
    -   [ ] Tablet
    -   [ ] Mobile

-   [ ] Person 2 (mandatory)

    Browser-Support

    -   [ ] Chrome
    -   [ ] Firefox
    -   [ ] Safari

    Responsiveness

    -   [ ] Desktop
    -   [ ] Tablet
    -   [ ] Mobile

Information transfer

-   [ ] Inform Ceres about changes in `TerraFormComponent` (optional: relevant for changes in this component)

Terra Basic Plugin

-   [ ] Adapt changes (optional: relevant for version updates, global design changes, l10n)

---

#### Must be done by every developer

Please inform a member of Terra (@plentymarkets/team-terra) to get the upper part of the checklist done (with urgency or deadline).

Documentation

-   [ ] Changelog (optional: relevant for every change which influences the API)
-   [ ] Example (updated or created)
-   [ ] JSDoc (optional: relevant for every change which influences the API)
-   [ ] [Google spreadsheet](https://docs.google.com/spreadsheets/d/1OINnux8TEoitV-qdAxqUQaf9oGI_fqFVwfRWenRFfhI/edit#gid=0) (relevant when deprecating public APIs or features)

Testing

-   [ ] Unit Test (updated or created)

### CODE REVIEW

File: src/lib/components/table/terra-table-data-source.ts [hunk 1]
```diff
@@ -53,25 +55,25 @@ import { TerraPagerInterface } from '../pager/data/terra-pager.interface';
  * }
  * ```
  */
-export abstract class TerraTableDataSource<T> extends DataSource<T> {
+export abstract class TerraTableDataSource<D, F = any> extends DataSource<D> {
```
Comment (pweyrich) [inline src/lib/components/table/terra-table-data-source.ts:58]: `F = any` oder doch lieber `F = unknown` ?
```suggestion
export abstract class TerraTableDataSource<D, F = unknown> extends DataSource<D> {
```

**Review by MFrank84 — APPROVED** [review]

File: src/lib/components/table/terra-table-data-source.ts (hunk 1 above)
Comment (TimWieder) [reply src/lib/components/table/terra-table-data-source.ts:58]: Das funktioniert leider nicht. Gibt nen TypeError

Comment (TimWieder) [reply src/lib/components/table/terra-table-data-source.ts:58]: Deswegen muss es leider F = any sein

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,9 @@
+# 10.0.0 (unreleased)
+
+### Features
+
+-   **table** - added type generic for filters
```
Comment (pweyrich) [inline CHANGELOG.md:5]:
```suggestion
-   **table** added another type generic for filter parameters to `TerraTableDataSource`.
```

**Review by pweyrich — APPROVED** [review]

**Review by MFrank84 — APPROVED** [review]

**Pull request merged**